src/Tickets/Commerce/Hooks.php

```diff
@@ -728,4 +731,17 @@ public function register_event_compatibility_hooks() {
 	public function include_assets_checkout_shortcode() {
 		Shortcodes\Checkout_Shortcode::enqueue_assets();
 	}
+
+	/**
+	 * Intercept the currency code option updates to set defaults according to the new selection.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $options The options being updated.
+	 *
+	 * @return array The options being updated.
+	 */
+	public function intercept_currency_symbol_change( $options  ) {
+		return tribe( Base_Commerce\Settings::class )->set_default_format_on_currency_change( $options );
```

Comment:
This will make us opinionated and setting what we think is better for a currency. We should allow users to use whatever they wish for the decimal and thousands separator.

Reply:
It is actually just intercepting whenever you switch between currencies to avoid any mistakes made by the user. They can always use custom formats as needed. Here is a screencast detailing the issue: https://d.pr/v/yz3Pt3

Reply:
Thanks for the video @rafsuntaskin. Right, that's what I'm saying. Imagine I install the plugin and go to set up Tickets Commerce. I realize the default value is USD, so I change it to Euro and I put 4 for the number of decimals, and `|` for the decimal separator. Once I hit save, those things won't be saved. 

The values that are going to be saved are the defaults, that we as a plugin in an opinionated way, decided to save. I understand where this is coming from, but I don't feel that we as a plugin should be changing the value (not even for the first time) of what a user is trying to set up for their plugin settings.

Reply:
@juanfra yes, your point is valid also. That is why we will need the format preview so that customers can exactly see the reflection.

Reply:
I have removed them now.